cmd/soroban-cli/src/commands/contract/invoke.rs

```diff
@@ -262,21 +263,24 @@ impl Cmd {
         Ok((function.clone(), spec, invoke_args, signers))
     }
 
-    pub async fn run(&self) -> Result<(), Error> {
-        let res = self.invoke().await?;
+    pub async fn run(&self, global_args: &global::Args) -> Result<(), Error> {
+        let res = self.invoke(global_args).await?;
         println!("{res}");
         Ok(())
     }
 
-    pub async fn invoke(&self) -> Result<String, Error> {
+    pub async fn invoke(&self, global_args: &global::Args) -> Result<String, Error> {
         if self.config.is_no_network() {
-            self.run_in_sandbox()
+            self.run_in_sandbox(global_args)
         } else {
-            self.run_against_rpc_server().await
+            self.run_against_rpc_server(global_args).await
         }
     }
 
-    pub async fn run_against_rpc_server(&self) -> Result<String, Error> {
+    pub async fn run_against_rpc_server(
```

Comment:
The RPC server when doing simulation should have access to the same budget that the sandbox execution has access to.

Why do we limit ourselves to outputting the `SorobanResource`, and not outputting the `Budget` for the simulated portion?

Reply:
It should have that yes. It will. But the RPC doesn't respond with that. So we could do that if we change the RPC API to expose that info, but that's probably a follow-up PR, if we want to do that.

Reply:
Which, frankly, if we're trying to provide that level of detail to the user, I'm not convinced the "Budget" object is the best way to do that. A profiling flame graph of heap allocation profile graph would be much more useful tools for improving contract performance, IMO.